posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,168 @@
+
+[.imageblock.img_border_light]
+image::/img/blog/pipeline-code-on-jenkins.png[Pipeline code directly on Jenkins,align="center"]
+
+* As plain text in a Jenkinsfile (a plain text file) in Git and mapping it to Jenkins. This is better for ensuring that your configuration is always under version control. If you have specific build and deployment configurations for separate environments, such as dev, staging, and production, you create a separate Jenkinsfile for each environment and store it in that environment-specific Git repository branches. You can use either the "Pipeline" or "Multibranch pipeline" type of job in this case.
+
+[.imageblock.img_border_light]
+image::/img/blog/pipeline-code-on-git.png[Pipeline code stored in a Jenkinsfile on Git,align="center"]
+
+== Building the cloud-native Java application with Open Liberty on Jenkins
+
+The following sample pipeline code builds your Java application code, packages it into a Docker container image, and pushes the container image to a remote container image repository, such as Docker Hub or an equivalent within your enterprise:
+
+[source]
+----
+ pipeline {
+     agent any
+      stages {
+       stage('Build') {
+                    steps {
+              checkout([$class: 'GitSCM', branches: [[name: '*/main']], extensions: [], userRemoteConfigs: [[credentialsId: ‘<git token>, url: 'https://github.com/liberty/app.git']]])
+                    sh '''
+                         mvn -U package
+                         docker login <remote-docker-image-repository-url> -u "${USERNAME}" -p “${PASSWORD}”
+                         docker build -t liberty-$<code identifier>:$<docker image version> .
+                         #eg: docker build -t liberty-app:v1.0 .
+                         docker tag liberty-$<code identifier>:$<docker image version> <remote-docker-image-repository-url>/<docker-repo-name>/liberty-$<code identifier>:$<docker image version>
+                         #eg: docker tag liberty-app:v1.0 docker.io/someid/liberty-app:v1.0
+                         docker push <remote-docker-image-repository-url>/<docker-repo-name>/liberty-$<code identifier>:$<docker image version>
+                         #eg: docker push docker.io/someid/liberty-app:v1.0  
+                       '''   
+                  }
+               }
+            }
+          }
+----
+
+Where:
+
+* _git-token_ is the personal access token you have generated from your Github account.
+* _remote-docker-image-repository-url_ is the location of the Docker image repository.
+* _username_ is your user name for the Docker image repository.
+* _password_ is your password for the Docker image repository.
+* _docker-image-version_ is the version number of your Docker image, a unique identifier.
+
+Use this link:/guides/containerize.html[sample Dockerfile] and customize it for your application.
```

Comment:
Consider the following:

"Build your container image using a Dockerfile or a Containerfile generated by Liberty Starter ([start.openliberty.io](http://start.openliberty.io/)) or by following the pattern described at https://github.com/OpenLiberty/ci.docker.   The Containerize guide (https://openliberty.io/guides/containerize.html) is a helpful resource that goes into more details on how to create a container image for applications running on Liberty."

Cc @lauracowen

Reply:
Done